core/good_key.go

```diff
@@ -54,37 +53,135 @@ func GoodKey(key crypto.PublicKey) error {
 	case *ecdsa.PublicKey:
 		return GoodKeyECDSA(*t)
 	default:
-		err := MalformedRequestError(fmt.Sprintf("Unknown key type %s", reflect.TypeOf(key)))
-		log.Debug(err.Error())
-		return err
+		return MalformedRequestError(fmt.Sprintf("Unknown key type %s", reflect.TypeOf(key)))
 	}
 }
 
 // GoodKeyECDSA determines if an ECDSA pubkey meets our requirements
 func GoodKeyECDSA(key ecdsa.PublicKey) (err error) {
-	log := blog.GetAuditLogger()
-	err = NotSupportedError("ECDSA keys not yet supported")
-	log.Debug(err.Error())
-	return
+	// Check the curve.
+	//
+	// The validity of the curve is an assumption for all following tests.
+	err = GoodCurve(key.Curve)
+	if err != nil {
+		return err
+	}
+
+	// Key validation routine adapted from NIST SP800-56A § 5.6.2.5.
```

Comment:
Note that in the current revision(Revision 2) of NIST SP800-56A from 2013 this moved to 5.6.2.3.2.
For reference: http://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar2.pdf